common/src/main/java/com/hartwig/serve/common/ensemblcache/EnsemblDataCache.java

```diff
@@ -11,12 +12,22 @@ public class EnsemblDataCache {
     @NotNull
     private final Map<String, List<GeneData>> genesPerChromosome;
     @NotNull
+    private final Map<String, Map<String, GeneData>> genesPerChromosomeAndGeneName;
+    @NotNull
     private final Map<String, List<TranscriptData>> transcriptsPerGeneId;
 
     public EnsemblDataCache(@NotNull final Map<String, List<GeneData>> genesPerChromosome,
             @NotNull final Map<String, List<TranscriptData>> transcriptsPerGeneId) {
         this.genesPerChromosome = genesPerChromosome;
         this.transcriptsPerGeneId = transcriptsPerGeneId;
+        this.genesPerChromosomeAndGeneName = new HashMap<>();
```

Comment:
Why keep all genes per chromosome? It seems gene names are unique over all chromosomes. And if they're not the current code already returned the first gene in the first chromosome that matched.

The only usage of this map is to speedup `findGeneDataByName`, which returns the first matching gene it finds. So it should be enough to keep one `Map<String, GeneData> genesPerName`, loop over all chromosomes once here and store the first gene found (e.g. `genesPerName.putIfAbsent(gene.geneName(), gene))`). Instead of looping over all chromosomes all the time in `findGeneDataByName` you can then let it return `genesPerName.get(geneNameToFind)` directly. The net result is the same: you'll return the first matching gene in the list of chromosomes.